zk/conn.go

```diff
@@ -842,6 +842,25 @@ func (c *Conn) ExistsW(path string) (bool, *Stat, <-chan Event, error) {
 	return exists, &res.Stat, ech, err
 }
 
+func (c *Conn) CancelEvent(evt <-chan Event) {
+	c.watchersLock.Lock()
+	defer c.watchersLock.Unlock()
+	for pathType, watchers := range c.watchers {
+		for idx, ch := range watchers {
+			if evt == ch {
+				close(ch)
+				c.watchers[pathType] = append(watchers[:idx], watchers[idx+1:]...)
+				return
+			}
+		}
+	}
+	select {
```

Comment:
should this select be wrapped in a forever loop in case there are multiple events? if so, then probably want to break out of the loop if the chan is closed (to avoid busylooping)